### ISSUE TITLE
Metrics: MSE, RMSE, MAE, RMSLE

### ISSUE THREAD

**Issue by justusschock** [PyTorchLightning/pytorch-lightning#1304]
## 🚀 Feature
Implement the following as metrics (grouped due to similarity):
- [ ] MSE
- [ ] RMSE
- [ ] MAE
- [ ] RMSLE

> Note: Probably it suffices for some of them to derive a new class from the loss variant in `torch.nn` and the `BaseMetric` class

**Comment by Borda** [comment]
@cuent ^^ mind type anything so I can assign it to you... :rabbit:

**Comment by cuent** [comment]
Thanks 😄 Will try to push tmrr!

**Comment by flxai** [comment]
For anyone landing here from a search engine: The situation has changed since 2020 and almost all are available in [TorchMetrics](https://torchmetrics.readthedocs.io/en/latest/), [an official PyTorch Lightning project to outsource metrics](https://www.pytorchlightning.ai/blog/torchmetrics-pytorch-metrics-built-to-scale).

* [MSE](https://torchmetrics.readthedocs.io/en/stable/regression/mean_squared_error.html)
* RMSE
  * `torch.sqrt(mse(y, y_hat))`
* [MAE](https://torchmetrics.readthedocs.io/en/stable/regression/mean_absolute_error.html)
* RMSLE
  * `torch.sqrt(msle(y, y_hat))`
  * Using [MSLE](https://torchmetrics.readthedocs.io/en/stable/regression/mean_squared_log_error.html)
